### ISSUE TITLE
Computed properties in destructured assignment

### ISSUE THREAD

**Issue by pmowrer** [palantir/tslint#1550]
### Bug Report

- __TSLint version__: `3.15.1`
- __TypeScript version__: `1.8.10`
- __Running TSLint via__:  CLI

#### TypeScript code being linted
The following computed properties destructuring assignment example is taken straight from [MDN's docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Computed_object_property_names_and_destructuring). 

```ts
let key = "z";
let { [key]: foo } = { z: "bar" };

console.log(foo); // "bar"
```

with `tslint.json` configuration: none passed (default CLI configuration).

#### Actual behavior
```bash
[2, 17]: Missing semicolon
[2, 32]: Missing semicolon
```

The TS compiler has no issues with it.

#### Expected behavior
No errors/warnings.

**Comment by JKillian** [comment]
@pmowrer I can't reproduce this - I tried running the version of TSLint you specified on the code sample you provided with `semicolon: [true, "always"]`, but  didn't get any errors. Can you double-check that this bug is reproducible in the minimal scenario you provided?

**Comment by pmowrer** [comment]
@JKillian Just tried on a secondary machine and can't reproduce. Sorry if I jumped the gun. Will re-open if it resurfaces.

**Comment by JKillian** [comment]
No worries! I've seen this kind of thing happen before when there's an old version of TypeScript in the mix somehow - that's likely what happened here
